osf/models/outcomes.py

```diff
@@ -72,11 +79,12 @@ class Outcome(ObjectIDMixin, EditableFieldsMixin, BaseModel):
     def primary_osf_resource(self):
         return self.artifact_metadata.get(artifact_type=ArtifactTypes.PRIMARY).identifier.referent
 
-    def add_artifact_by_pid(self, pid, artifact_type, pid_type='doi', create_identifier=False):
-        return self.artifacts.through.objects.create_for_identifier_value(
-            outcome=self,
-            pid_value=pid,
-            pid_type=pid_type,
-            artifact_type=artifact_type,
-            create_identifier=create_identifier
+    def log_artifact_change(self, action, artifact, api_request, **log_params):
+        nodelog_action = NODE_LOGS_FOR_OUTCOME_ACTION[action]
+        nodelog_params = {'artifact_id': artifact._id, **log_params}
+
+        self.primary_osf_resource.add_log(
```

Comment:
We typically don't add new logs to `Registration` objects (the only valid `primary_osf_resource` type for now), and they aren't displayed anywhere. If we add a display for them later, or when `Node` becomes a permissible `primary` resource type, it make cause issues that neither `should_hide` is `True`, nor does `logsActionList.json` have entries for the new logs.

Reply:
I'm would need a reeeaalllly good reason to  keep logging Registration-specific activity to the `registered_node`, especially now that we're well into the world of No Project Registrations. That said I'm fine updating `logActionsList.json`.